pkg/controller/master-controller-manager/kubeone/controller.go

```diff
@@ -511,3 +707,379 @@ func generateConfigMap(name, namespace, action string) *corev1.ConfigMap {
 		},
 	}
 }
+
+func (r *reconciler) upgradeCluster(ctx context.Context, externalCluster *kubermaticv1.ExternalCluster) error {
+	r.log.Debugw("Generate kubeone pod to upgrade kubeone", "Cluster", externalCluster.Name)
```

Comment:
Have you thought about simply adding the cluster name to the logger and then using that logger, like

```go
logger := r.log.With("cluster", externalCluster.Name)

// yadda yadda yadda

r.upgradeCluster(ctx, logger, externalCluster) // <--- doesn't now need to constantly add the same fields to the logger
```